WooCommerce/Classes/ViewRelated/Products/Edit Product/BottomSheetListSelector/ProductFormBottomSheetActionsFactory.swift

```diff
@@ -0,0 +1,17 @@
+import Yosemite
+
+struct ProductFormBottomSheetActionsFactory {
+    /// Retruns an array of actions that are visible in the product form bottom sheet.
+    static func actions(product: Product, isEditProductsRelease2Enabled: Bool, isEditProductsRelease3Enabled: Bool) -> [ProductFormBottomSheetAction] {
+        let shouldShowShippingSettingsRow = product.isShippingEnabled
+        let shouldShowCategoriesRow = isEditProductsRelease3Enabled
+        let shouldShowShortDescriptionRow = isEditProductsRelease2Enabled
+        let actions: [ProductFormBottomSheetAction?] = [
+            .editInventorySettings,
+            shouldShowShippingSettingsRow ? .editShippingSettings: nil,
+            shouldShowCategoriesRow ? .editCategories: nil,
+            shouldShowShortDescriptionRow ? .editBriefDescription: nil
+        ]
+        return actions.compactMap({ $0 }).filter({ $0.isVisible(product: product) })
```

Comment:
I wonder if we should also encapsulate the `isVisible` logic inside this `Factory`. 🤔 Though that is also eventually used within `DefaultProductFormTableViewModel.settingsRows`. This makes me think that all these visibility logic should be merged too. 

I wonder if we've outgrown the current architecture (expected and normal) and we need to do a review at some point in the future.

Reply:
yea I think it'd be nice if we can merge the logic for making the bottom sheet actions and settings rows since they are based on the same data (product editing categories), maybe having both created from the same factory (a different name like `ProductFormActionsFactory`). what do you think? I created an issue for tech debt https://github.com/woocommerce/woocommerce-ios/issues/2317 to track 🙂